src/test/java/io/github/w7mike/logic/ProjectsServiceTest.java

```diff
@@ -105,4 +130,59 @@ private JobConfigurationProperties configurationReturning(final boolean result)
         when(mockProperties.getTemplate()).thenReturn(mockTemplate);
         return mockProperties;
     }
+
+    private InMemoryGroupRepository inMemoryGroupRepository(){
+        return new InMemoryGroupRepository() {
+
+        };
+    }
+
+    private static class InMemoryGroupRepository implements JobGroupsRepository{
+
+        private Integer index = 0;
+        private Map<Integer, JobGroups> map = new HashMap<>();
+
+        public int count(){
+            return map.values().size();
+        }
+
+
+        @Override
+        public List<JobGroups> findAll() {
+            return new ArrayList<>(map.values());
+        }
+
+        @Override
+        public Optional<JobGroups> findById(final Integer id) {
+            return Optional.ofNullable(map.get(id));
+        }
+
+        @Override
+        public JobGroups save(final JobGroups entity) {
+            if (entity.getId() == 0){
+                try {
+                    JobGroups.class.getDeclaredField("id").set(entity, ++index);
+                } catch (NoSuchFieldException | IllegalAccessException e) {
+                    throw new RuntimeException();
+                }
+            }
+            map.put(entity.getId(), entity);
+            return entity;
+        }
+
+        @Override
+        public void deleteById(final Integer id) {
+
+        }
+
+        @Override
+        public boolean existsByCompleteIsFalseAndProjects_Id(final Integer projectId) {
+            return map.values().stream()
+                    .filter(jobGroups -> !jobGroups.isComplete())
+                    .anyMatch(jobGroups -> jobGroups.getProjects() != null && jobGroups.getProjects().getId() == projectId);
+        }
+
+    }
+
+
 }
```

Comment:
Tiny thing, but there is old good practice to end every file with an empty new line.